### ISSUE TITLE
Support output arrays of numpy.ndarray type on ufuncs

### ISSUE THREAD

**Issue by manopapad** [nv-legate/cunumeric#342]
This already works for some other functions, e.g.:

```
x = cn.ones((4,4))
y = cn.ones((4,4))
z = np.zeros((4,4))
cn.dot(x, y, out=z)
print(z)
```

prints:

```
[[4. 4. 4. 4.]
 [4. 4. 4. 4.]
 [4. 4. 4. 4.]
 [4. 4. 4. 4.]]
```

but not for ufuncs:

```
x = cn.arange(4)
y = cn.arange(4)
z = np.zeros(4, dtype=np.int64)
cn.multiply(x, y, out=z)
print(z)
```

throws:

```
TypeError: return arrays must be of ArrayType
```

**Comment by magnatelee** [comment]
@manopapad I assume this is fixed by #315. Any objection to closing this?

**Comment by manopapad** [comment]
No, I'm closing this issue
